tds/src/test/java/thredds/server/notebook/TestNotebookService.java

```diff
@@ -0,0 +1,60 @@
+package thredds.server.notebook;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.checkerframework.checker.units.qual.C;
+import org.json.JSONObject;
+import org.junit.Test;
+import thredds.client.catalog.Catalog;
+import thredds.client.catalog.Dataset;
+import ucar.nc2.constants.FeatureType;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TestNotebookService {
+
+  @Test
+  public void testNotebookRegistration()
+      throws NotebookMetadata.InvalidJupyterNotebookException, FileNotFoundException, URISyntaxException {
+
+    final String test_file = "src/test/data/testNotebookMetadata.json";
+    NotebookMetadata nbData = new NotebookMetadata(new File(test_file));
+
+    // dataset that matches by ID
+    Map fldsWithId = new HashMap();
```

Comment:
to avoid a warning, maybe its nice to parametrize the map
```suggestion
    Map<String, Object> fldsWithId = new HashMap<>();
```